src/components/Layout/DrawerAppBar/DrawerAppBar.js

```diff
@@ -77,12 +123,34 @@ const StyledInputBase = styled(InputBase)(({ theme }) => ({
 
 function DrawerAppBar(props) {
     const navigate = useNavigate();
+    const dispatch = useDispatch();
     const [mobileOpen, setMobileOpen] = React.useState(false);
+    const {user} = useSelector(state => state.authReducer);
 
     const handleDrawerToggle = () => {
         setMobileOpen(!mobileOpen);
     };
 
+    const navItems = navigationItems.filter(item => item.role.includes(user?user.type:GUEST));
```

Comment:
```suggestion
    const navItems = user ? navigationItems.filter(item => item.role.includes(user.type)) : navigationItems.filter(item => item.role.includes(GUEST))
```
small performance change - you were querying `user ? ` in each iteration, this way it only queries it once. Also, spacing please :)